### PR TITLE
fix bash syntax and error handling

### PR DESCRIPTION
**Pull request by odra**
* fixes error handing
* use posix compliant conditional syntax

### CODE REVIEW

File: image/tools/lib/encryption/gpg.sh [hunk 1]
```diff
@@ -1,6 +1,6 @@
 function check_encryption_enabled {
     local result=$(oc get secret -n ${ENCRYPTION_SECRET_NAMESPACE} ${ENCRYPTION_SECRET_NAME} -o template --template='{{.metadata.name}}')
-    if [[ "$result" == "${ENCRYPTION_SECRET_NAME}" ]]; then
+    if [ "$result" == "${ENCRYPTION_SECRET_NAME}" ]; then
```
Comment (matskiv) [inline image/tools/lib/encryption/gpg.sh]: You used `-eq` in other conditions, why not here?

Comment (odra) [reply image/tools/lib/encryption/gpg.sh]: whoops forgot to change it

**Review by jessesarn — APPROVED** [review]
Looks good

**Comment by maleck13** [comment]
We will need an upgrade for this in the 1.4 upgrade playbook
Do we also need a change to installation to use this image?

**Comment by maleck13** [comment]
Ok lets create a follow up for the upgrade piece and then get this merged and cherry-picked to the 1.4 branch

**Comment by matskiv** [comment]
I finished the testing. Everything works(and fails) as expected.
This improvement actually helped to uncover an issue with 3Scale 2.5 postgresql backup - https://issues.jboss.org/browse/INTLY-2023

We will have to do a new release of backup image, and [bump version in the installer](https://github.com/integr8ly/installation/blob/edef60c1880aec644c4464c6c7787c79fb586093/inventories/group_vars/all/manifest.yaml#L105).

**Comment by matskiv** [comment]
I finished testing. Everything works(and fails) as expected. 👍 

Improvement from this PR actually helped uncover another issue - https://issues.jboss.org/browse/INTLY-2023

For both installation and upgrade, we will have to do a new release of the backup image, and then bump [the version in the installer](https://github.com/integr8ly/installation/blob/edef60c1880aec644c4464c6c7787c79fb586093/inventories/group_vars/all/manifest.yaml#L105)

**Comment by matskiv** [comment]
Yeah, good point about the upgrade 👍 
Maybe we can write something generic, that will update backup image version on each upgrade, so we don't have to do it each release?

PR looks good to me, I am building the image now, and testing it on my cluster in a moment.

**Comment by odra** [comment]
does it need a cherry pick into the 1.4 branch?

**Comment by maleck13** [comment]
yes also needs a follow up to deal with the upgrade

**Pull request merged**